### PR TITLE
Clean Up Utility Functions (Resolves #36)

### LINKED ISSUE #36: Clean up utility functions
The file main.js currently has a few general-ish utility functions that should be moved elsewhere. I would suggest creating a new file called utility.js, and moving the functions there. The particular functions I'm talking about are:
- `timeSince`
- `resizeArray`
- `arraysEqual`
- `testResizeArray`

It is _not_ in the scope of this issue to turn `testResizeArray` into a proper test. Here are some important tips:
- Make sure to follow the [Github Flow](https://guides.github.com/introduction/flow/); if you do your work in the main branch, Github will reject it
- If you have any JavaScript questions, you can ask me or look at the great [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference)
- Make sure to include `'use strict'` as the first line to enable [strict mode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode)
- Remember to run the linter and tests frequently with `npm test`. If you get a linter error, it should include a link to an explanation of the linter rule and perhaps why it's useful. If you're still confused, or think it should be changed, create an issue here on Github or contact me.
- In particular, the linter is pretty picky about whitespace and it requires you to always use `const` or `let`, never `var`
- This project uses the [CommonJS](https://nodejs.org/docs/latest/api/modules.html) module system
- At the end of the file, you'll want to write a `module.exports` statements that might look something like this:

```
module.exports = {
  timeSince: timeSince,
  resizeArray: resizeArray
};
```
- Note that you do not need to export `arraysEqual` or `testResizeArray`, as they are only used internally
- You will also need to import the respective functions in `main.js`. That might look something like this:

```
const timeSince = require('./utility.js').timeSince;
const resizeArray = require('./utility.js').resizeArray;
```

### PR DESCRIPTION
**Pull request by sawyerWeld**
moves utility methods from main.js to a new utility.js

### CODE REVIEW

**Comment by jackie-scholl** [comment]
Build failing, ping me when it's ready for review.

**Comment by ghost** [comment]
I've updated my changes, and it passes the checks this time. utilities.js was not staged for commits before.

File: src/main/resources/js/utility.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  console.assert(arraysEqual(resizeArray([0, 0, 1, 0, 0, 0], 0, 100, zeroTester,
+        zeroFiller), [0, 0, 1, 0]));
+  console.assert(arraysEqual(resizeArray([1, 0, 0, 6], 0, 100, zeroTester,
+        zeroFiller), [1, 0, 0, 6, 0]));
+};
+/* Wow. */
+testResizeArray();
+
+module.exports = {
+  arraysEqual: arraysEqual,
```
Comment (jackie-scholl) [inline src/main/resources/js/utility.js]: Should this function really be exported? Why?

File: src/main/resources/js/utility.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  testResizeArray: testResizeArray,
```
Comment (jackie-scholl) [inline src/main/resources/js/utility.js]: Same here

File: src/main/resources/js/utility.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    now that we have those. */
+const testResizeArray = function() {
+  const zeroTester = (x) => (x !== 0);
+  const zeroFiller = () => (0);
+  console.assert(arraysEqual(resizeArray([0, 0, 1, 0, 0, 0], 0, 100, zeroTester,
+        zeroFiller), [0, 0, 1, 0]));
+  console.assert(arraysEqual(resizeArray([1, 0, 0, 6], 0, 100, zeroTester,
+        zeroFiller), [1, 0, 0, 6, 0]));
+};
+/* Wow. */
```
Comment (jackie-scholl) [inline src/main/resources/js/utility.js:86]: There should probably be a blank line above this

File: src/main/resources/js/utility.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  interval = Math.floor(seconds / 3600);
+  if (interval > 1) {
+    return interval + ' hours';
+  }
+  interval = Math.floor(seconds / 60);
+  if (interval > 1) {
+    return interval + ' minutes';
+  }
+  return 'just now';
+  //return Math.floor(seconds) + ' seconds';
```
Comment (jackie-scholl) [inline src/main/resources/js/utility.js]: Does this commented out line need to stay in?

File: src/main/resources/js/main.js [hunk 5]
```diff
@@ -173,6 +149,7 @@ const ContractElement = React.createClass({
   }
 });
 
+
```
Comment (jackie-scholl) [inline src/main/resources/js/main.js]: Why was this line removed?

File: src/main/resources/js/main.js [hunk 6]
```diff
@@ -85,6 +87,7 @@ const FullPage = React.createClass({
...
```
Comment (jackie-scholl) [inline src/main/resources/js/main.js]: Why was this line added?

**Review by jackie-scholl — CHANGES_REQUESTED** [review]

**Comment by jackie-scholl** [comment]
Resolves  #36

File: src/main/resources/js/main.js [hunk 7]
```diff
@@ -577,4 +493,4 @@ console.log(`NODE_ENV: ${process.env.NODE_ENV}`);
 ReactDOM.render(
   <FullPage />,
   document.getElementById('content')
-);
+);
```
Comment (jackie-scholl) [inline src/main/resources/js/main.js]: I'd think files should probably end with a newline. See, for example, http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline. If you'd rather, I could accept this and we could fix it up as part of #48.

File: src/main/resources/js/utility.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        zeroFiller), [1, 0, 0, 6, 0]));
+};
+
+/* Wow. */
+testResizeArray();
+
+module.exports = {
+  resizeArray: resizeArray,
+  timeSince: timeSince
+};
```
Comment (jackie-scholl) [inline src/main/resources/js/utility.js]: Same as above

**Comment by jackie-scholl** [comment]
Thanks for fixing all of the previously mentioned issues, if you could just add line breaks to the end of those two files that'd be great, thanks.

File: src/main/resources/js/utility.js (hunk 8 above)
Comment (sawyerWeld) [reply src/main/resources/js/utility.js]: all files have a new line at the end

**Review by jackie-scholl — APPROVED** [review]

**Pull request merged**